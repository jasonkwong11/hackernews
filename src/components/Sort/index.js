import React from 'react';
import classnames from 'classnames';
import Button from '../Button'

const Sort = ({
  sortKey,
  onSort,
  activeSortKey,
  children
}) => {
  const sortClass = classnames(
    'button-inline',
    { 'button-active': sortKey === activeSortKey }
  );

  return (
    <Button
      onClick={() => onSort(sortKey)}
      className={sortClass}
    >
      {children}
    </Button>
  )
}

export default Sort
