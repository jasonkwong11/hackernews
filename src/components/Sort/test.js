import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Sort from './index';


describe('Sort', () => {
  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Sort/>, div);
  });

  test('snapshots', () => {
    const component = renderer.create(
      <Sort/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})