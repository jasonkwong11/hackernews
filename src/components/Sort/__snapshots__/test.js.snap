// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort snapshots 1`] = `
<button
  className="button-inline button-active"
  onClick={[Function]}
  type="button"
/>
`;
